File: fullstack/python/price_calculator/app/cart/managers/base_price_manager.py
from collections import defaultdict
from typing import Dict, List

from app.cart.models.base_price import ProductBasePrice
from app.common.utils import read_json


class ProductBasePriceManager:
    data: Dict[str, List[ProductBasePrice]]

    def __init__(self, json_file=None):
        self.data = self.read_json(json_file)

    @staticmethod
    def read_json(json_file):
        data = defaultdict(list)
        raw_data = read_json(json_file)

        for raw_item in raw_data:
            product_base_price = ProductBasePrice.from_dict(raw_item)
            data[product_base_price.product_type].append(product_base_price)

        return data

    def lookup(self, product_type, options=None) -> 'ProductBasePrice':
        raise NotImplementedError

File: fullstack/ruby/price_calculator/lib/cart/managers/base_price_manager.rb
require 'json'

require_relative '../../cart/models/product_base_price'

class BasePriceManager
  attr_accessor :data

  def initialize(json_file:)
    @data = JSON.parse(File.read(json_file))
  end

  def lookup(product_type:, options: nil)
    raise NotImplementedError
  end
end